filters/diag/diag.go

```diff
@@ -417,3 +457,103 @@ func (t *throttle) Response(ctx filters.FilterContext) {
 	rsp := ctx.Response()
 	rsp.Body = t.goThrottle(rsp.Body, true)
 }
+
+func (j *jitter) Name() string {
+	switch j.typ {
+	case normalDist:
+		return "normal"
+	case gaussianDist:
+		return "gaussian"
+	case randomDist:
+		return "random"
+	}
+	return "unknown"
+}
+
+func (j *jitter) CreateFilter(args []interface{}) (filters.Filter, error) {
+	var (
+		delay     time.Duration
+		interval  time.Duration
+		mu, sigma float64
+		err       error
+	)
+
+	if !(len(args) == 2 || len(args) == 4) {
+		return nil, errors.New("filter requires 2 or 4 parameters")
+	}
+
+	if delay, err = parseDuration(args[0]); err != nil {
+		return nil, fmt.Errorf("failed to parse duration delay %v: %w", args[0], err)
+	}
+
+	if interval, err = parseDuration(args[1]); err != nil {
+		return nil, fmt.Errorf("failed to parse duration interval %v: %w", args[1], err)
+	}
+
+	switch j.typ {
+	case normalDist:
+		mu = 0
+		sigma = 1
+
+	case gaussianDist:
+		if len(args) != 4 {
+			return nil, errors.New("gaussian jitter filter requires 4 parameters")
+		}
+
+		s, ok := args[2].(string)
+		if !ok {
+			return nil, fmt.Errorf("failed to convert mu to string %v", args[2])
+		}
+
+		mu, err = strconv.ParseFloat(s, 32)
+		if err != nil {
+			return nil, fmt.Errorf("failed to parse float mu %v: %w", args[2], err)
+		}
+
+		s, ok = args[3].(string)
+		if !ok {
+			return nil, fmt.Errorf("failed to convert sigma to string %v", args[3])
+		}
+
+		sigma, err = strconv.ParseFloat(s, 32)
+		if err != nil {
+			return nil, fmt.Errorf("failed to parse float sigma %v: %w", args[3], err)
+		}
+
+	case randomDist:
+	default:
+		return nil, errors.New("distribution does not exist")
+	}
+
+	return &jitter{
+		typ:      j.typ,
+		delay:    delay,
+		interval: interval,
+		mu:       mu,
+		sigma:    sigma,
+	}, nil
+}
+
+func (j *jitter) Request(filters.FilterContext) {
+	d := j.delay
+
+	switch j.typ {
+	case randomDist:
+		r := rand.Float64() - 0.5 // +/- \epsilon
+		f := r * float64(j.interval)
+		d += time.Duration(int64(f))
+	case normalDist:
+		r := rand.NormFloat64()
+		f := r * float64(j.interval)
+		d += time.Duration(int64(f))
+	case gaussianDist:
+		r := rand.NormFloat64()
+		r = r*j.sigma + j.mu
+		f := r * float64(j.interval)
+		d += time.Duration(int64(f))
+	}
```

Comment:
Maybe lets just have one "normal" with sigma and mu (which is called `gaussianDist` here)
and let users to specify (sigma=1, mu=0) if they like.
https://en.wikipedia.org/wiki/Normal_distribution
https://stats.stackexchange.com/questions/55962/what-is-the-difference-between-a-normal-and-a-gaussian-distribution

Reply:
I am fine with that.

Reply:
done